src/azure-cli-core/azure/cli/core/extension/operations.py

```diff
@@ -213,18 +213,20 @@ def _install_deps_for_psycopg2():  # pylint: disable=too-many-statements
         distname, _ = get_linux_distro()
         distname = distname.lower().strip()
         if installer == 'DEB' or any(x in distname for x in ['ubuntu', 'debian']):
-            exit_code = subprocess.call(['dpkg', '-s', 'gcc', 'libpq-dev', 'python3-dev'], stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL)
-            if exit_code != 0:
-                logger.warning('This extension depends on gcc, libpq-dev, python3-dev and they will be installed first.')
-                apt_update_cmd = 'apt-get update'.split()
-                apt_install_cmd = 'apt-get install -y gcc libpq-dev python3-dev'.split()
-                if os.geteuid() != 0:  # pylint: disable=no-member
-                    apt_update_cmd.insert(0, 'sudo')
-                    apt_install_cmd.insert(0, 'sudo')
-                exit_code = subprocess.call(apt_update_cmd, True)
-                if exit_code == 0:
-                    logger.debug("Install dependencies with '%s'", " ".join(apt_install_cmd))
-                    subprocess.call(apt_install_cmd, True)
+            from azure.cli.core.util import in_cloud_console
+            if not in_cloud_console():
```

Comment:
The easiest code is to put this at the beginning of `_install_deps_for_psycopg2` at Line 185:

```py
if in_cloud_console():
    return
```

Reply:
That makes sense. Made the change accordingly. Thank you

Reply:
Please put these lines at L185.

Reply:
I have made the change.